app/Http/Controllers/V2/Terrafund/TerrafundClipGeometryController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Http\Controllers\V2\Terrafund;
+
+use App\Helpers\CreateVersionPolygonGeometryHelper;
+use App\Helpers\GeometryHelper;
+use App\Helpers\PolygonGeometryHelper;
+use App\Http\Controllers\Controller;
+use App\Models\V2\Sites\CriteriaSite;
+use App\Services\PolygonService;
+use App\Services\PythonService;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\Log;
+
+class TerrafundClipGeometryController extends Controller
+{
+    public function clipOverlappingPolygonsBySite(string $uuid)
+    {
+        $polygonUuids = GeometryHelper::getSitePolygonsUuids($uuid)->toArray();
+
+        return $this->processClippedPolygons($polygonUuids);
+    }
+
+    public function clipOverlappingPolygons(string $uuid)
+    {
+        $polygonOverlappingExtraInfo = CriteriaSite::where('polygon_id', $uuid)
+            ->where('criteria_id', PolygonService::OVERLAPPING_CRITERIA_ID)
```

Comment:
create a scope in the CriteriaSite Model to only pass a CriteriaID as a parameter

Reply:
using a already created scope now